### PR TITLE
auth 

### PR DESCRIPTION
**Pull request by manthankumbhar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,14 +3,22 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@testing-library/jest-dom": "^5.16.2",
-    "@testing-library/react": "^12.1.3",
+    "@emotion/react": "^11.7.1",
+    "@emotion/styled": "^11.6.0",
+    "@mui/material": "^5.3.0",
+    "@testing-library/jest-dom": "^5.16.1",
```
Comment (kavadithya) [inline package.json]: move testing packages to devDependencies?

File: src/Components/Home/Home.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { useNavigate } from "react-router";
+import { Link } from "react-router-dom";
+import auth from "../../hoc/auth";
+import img from "../../UI/under_construction.png";
+import logo from "../../UI/logo.svg";
+import "./Home.scss";
+
+export default function Home() {
+  const navigate = useNavigate();
+  const logout = (e) => {
```
Comment (kavadithya) [inline src/Components/Home/Home.js]: add useCallback

Comment (kavadithya) [reply src/Components/Home/Home.js]: it's used to cache functions - improves performance

File: src/Components/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { useNavigate } from "react-router";
+import { Link } from "react-router-dom";
+import auth from "../../hoc/auth";
```
Comment (kavadithya) [inline src/Components/Home/Home.js]: don't use relative paths anywhere if possible - set a root path and use absolute paths from there;

Comment (kavadithya) [reply src/Components/Home/Home.js]: do this everywhere - not repeating comment in other places.

File: src/Components/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import "./Login.scss";
+import logo from "../../UI/logo.svg";
+import { Link, useNavigate, Navigate } from "react-router-dom";
+import axios from "axios";
+import auth from "../../hoc/auth";
+import { Alert, CircularProgress, Snackbar } from "@mui/material";
+
+export default function Login(props) {
+  const [email, setEmail] = useState("");
```
Comment (kavadithya) [inline src/Components/Login/Login.js:11]: why empty string? use null instead

File: src/Components/Login/Login.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState("");
+  const [snackbarSeverity, setsnackbarSeverity] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const submitResetEmail = (e) => {
+    setLoading(true);
+    e.preventDefault();
+    axios
```
Comment (kavadithya) [inline src/Components/Login/Login.js]: don't use promises - use async / await instead

File: src/Components/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+import axios from "axios";
+import auth from "../../hoc/auth";
+import { Alert, CircularProgress, Snackbar } from "@mui/material";
+
+export default function Login(props) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordReset, setPasswordReset] = useState(false);
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(false);
```
Comment (kavadithya) [inline src/Components/Login/Login.js]: what is open? from reading the code, I can't understand what this variable is for; rename variable

File: src/Components/Login/Login.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+  const [snackbarMessage, setSnackbarMessage] = useState("");
```
Comment (kavadithya) [inline src/Components/Login/Login.js:16]: null initialization everywhere - unless some library prohibits it

File: src/Components/Login/Login.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        setsnackbarSeverity("error");
+        setSnackbarMessage(err.response.data["error"]);
+        setLoading(false);
+      });
+  };
+
+  const submitLogin = (e) => {
+    setLoading(true);
+    e.preventDefault();
+    axios
```
Comment (kavadithya) [inline src/Components/Login/Login.js]: async / await everywhere

File: src/Components/Login/Login.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+
+  const submitResetEmail = (e) => {
+    setLoading(true);
+    e.preventDefault();
+    axios
+      .post(`${process.env.REACT_APP_SERVER_LINK}/reset-password`, {
+        email: email,
+      })
+      .then((res) => {
+        if (res.status === 200) {
```
Comment (kavadithya) [inline src/Components/Login/Login.js:45]: what happens if res.status != 200? case not handled afaict

File: src/Components/Login/Login.js [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+  const submitLogin = (e) => {
+    setLoading(true);
+    e.preventDefault();
+    axios
+      .post(`${process.env.REACT_APP_SERVER_LINK}/signin`, {
+        email: email,
+        password: password,
+      })
+      .then((res) => {
+        if (res.status === 200) {
```
Comment (kavadithya) [inline src/Components/Login/Login.js:72]: what happens if res.status != 200? case not handled afaict

File: src/Components/Login/Login.js [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+      })
+      .catch((err) => {
+        setOpen(true);
+        setsnackbarSeverity("error");
+        setSnackbarMessage(err.response.data["error"]);
+        setLoading(false);
+      });
+  };
+
+  const handleClose = (event, reason) => {
```
Comment (kavadithya) [inline src/Components/Login/Login.js]: usecallback

File: src/Components/Login/Login.js [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+        >
+          {snackbarMessage}
+        </Alert>
+      </Snackbar>
+      <Link to="/" className="login__logo">
+        <img src={logo} alt="logo" />
+      </Link>
+      <div className="login__content">
+        <h1 className="login__content--header">Log in</h1>
+        {passwordReset ? (
```
Comment (kavadithya) [inline src/Components/Login/Login.js]: move to different components to make it more readable

File: src/Components/Login/Login.js [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+  };
+
+  const handleClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setOpen(false);
+  };
+
+  return (
```
Comment (kavadithya) [inline src/Components/Login/Login.js:179]: entire block is toooo long - always split up to components to improve readability

Comment (kavadithya) [reply src/Components/Login/Login.js:179]: will review this block after splitting to components

File: src/Components/Signup/Signup.js [hunk 14]
```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import "./Signup.scss";
+import logo from "../../UI/logo.svg";
+import { Link, Navigate, useNavigate } from "react-router-dom";
+import axios from "axios";
+import auth from "../../hoc/auth";
+import { Alert, CircularProgress, Snackbar } from "@mui/material";
+
+export default function Signup() {
```
Comment (kavadithya) [inline src/Components/Signup/Signup.js:10]: repeat some comments from Login - about initialization, naming variables etc.

Comment (kavadithya) [inline src/Components/Signup/Signup.js:10]: too much repeated code here

File: src/Components/Signup/Signup.js [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+        {email ? (
+          <div>
+            <label className="signup__content--label">Password</label>
+            <input
+              type="password"
+              name="password"
+              required
+              className="signup__content--input"
+              value={password}
+              onChange={(e) => setPassword(e.target.value)}
```
Comment (kavadithya) [inline src/Components/Signup/Signup.js]: don't use functions directly like this; create a separate function for this, use useCallback, and add that function name here

File: src/hoc/PrivateRoute.js [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  var isRefreshTokenExpired = isExpired(refreshToken);
+  if (isAccessTokenExpired === false) {
+    return true;
+  } else if (isRefreshTokenExpired === false) {
+    axios
+      .post(`${process.env.REACT_APP_SERVER_LINK}/refresh-token`, {
+        refreshToken: refreshToken,
+      })
+      .then((res) => {
+        if (res.status === 200) {
```
Comment (kavadithya) [inline src/hoc/PrivateRoute.js:19]: error case not handled

Comment (kavadithya) [reply src/hoc/PrivateRoute.js:19]: server error I mean

File: src/hoc/PrivateRoute.js [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+    auth.logout(() => {
+      localStorage.removeItem("accessToken");
+      localStorage.removeItem("refreshToken");
+    });
+  }
+}
+
+function PrivateRoute({ Component, ...rest }) {
+  const isAuth = auth.isAuthenticated();
+  return isAuth ? (
```
Comment (kavadithya) [inline src/hoc/PrivateRoute.js]: too many if / else conditions nested using ?:
don't do that; 
use proper if/else conditions and multiple return statements to make it more readable

File: src/hoc/PrivateRoute.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+    <Outlet />
+  ) : localStorage.accessToken && localStorage.refreshToken ? (
+    CheckToken(localStorage.accessToken, localStorage.refreshToken) ? (
+      <Outlet />
+    ) : (
+      <div>
+        <p style={{ fontSize: "35px", fontWeight: "900" }}>
```
Comment (kavadithya) [inline src/hoc/PrivateRoute.js]: css not separate file for this? unlike other files

**Review by kavadithya — CHANGES_REQUESTED** [review]
Home.js, Login.js & Signup.js have major comments; 
minor comments in other places

File: src/Components/Login/Login.js (hunk 4 above)
Comment (manthankumbhar) [reply src/Components/Login/Login.js:11]: tried but react gives out an error saying the value prop in the input file shouldn't be kept null, go for empty string instead

File: src/Components/Login/Login.js (hunk 7 above)
Comment (manthankumbhar) [reply src/Components/Login/Login.js:16]: every library needs empty string, not accepting null

File: src/Components/Login/Login.js (hunk 9 above)
Comment (manthankumbhar) [reply src/Components/Login/Login.js:45]: I tried adding case check for 400 but in they end up going to catch error, errors are automatically shifted to catch block and cannot be handled in that particular block

File: src/Components/Home/Home.js (hunk 3 above)
Comment (kavadithya) [reply src/Components/Home/Home.js]: https://stackoverflow.com/questions/41993461/import-react-components-with-absolute-path

File: src/Components/Login/Login.js (hunk 9 above)
Comment (kavadithya) [reply src/Components/Login/Login.js:45]: ok

**Review by kavadithya — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 19]
```diff
@@ -1,9 +1,33 @@
+import { Route, Routes } from "react-router";
 import "./App.scss";
```
Comment (kavadithya) [inline src/App.js]: avoid if possible

File: src/Components/Home/Home.js [hunk 20]
```diff
@@ -0,0 +1,30 @@
+import React, { useCallback } from "react";
+import { useNavigate } from "react-router";
+import { Link } from "react-router-dom";
+import auth from "hoc/auth";
+import img from "UI/under_construction.png";
+import logo from "UI/logo.svg";
+import "./Home.scss";
```
Comment (kavadithya) [inline src/Components/Home/Home.js]: fix

**Review by kavadithya — APPROVED** [review]
minor comment - please fix & squash merge to master

**Pull request closed without merging**